actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -2468,9 +2468,21 @@ def submit(value = nil, options = {})
       #   #      <strong>Ask me!</strong>
       #   #    </button>
       #
+      #   button do |text|
+      #     content_tag(:strong, text)
+      #   end
+      #   # => <button name='button' type='submit'>
+      #   #      <strong>Create post</strong>
+      #   #    </button>
+      #
       def button(value = nil, options = {}, &block)
         value, options = nil, value if value.is_a?(Hash)
         value ||= submit_default_value
+
+        if block_given?
+          block = proc { yield(value) }
```

Comment:
@rafaelfranca surprisingly, I was unable to get this working with `value = yield(value)`, in any form.

Reply:
Is there a more idiomatic way to invoke the block parameter with an argument?